### ISSUE TITLE
Add a predictor method to return more than one possible sequence

### ISSUE THREAD

**Issue by juan-cb** [IBM/pytorch-seq2seq#113]
Would be possible to add to this library a `predictor_n` (or to modify the current `predictor`) method to return more than one sequence as result? I think it would be a great tool to have when using beam search (with `TopKDecoder`).

I coded a first attempt to do that (it seems to work, https://github.com/juan-cb/pytorch-seq2seq/commit/442431001b122fa15c4b6476a9d7411570f53f20), but I'm not sure if it is the best way to implement that or is completely correct. The desired behavior is to return the `n` most probable sequences given an `src_seq`.

Thanks in advance

**Comment by kylegao91** [comment]
I think it's nice feature to have and thanks for the attempted implementation.  

I noticed that you use `other["sequence"]`, which contains sequences for the input batch rather than the beams of each input.  Instead, as noted [here](https://ibm.github.io/pytorch-seq2seq/public/models.html#seq2seq.models.TopKDecoder.TopKDecoder), the `other['topk_sequence']` should be used to get the beam sequences.

**Comment by juan-cb** [comment]
Thank you @kylegao91 for the reply and the clarification about the `other['topk_sequence']`. 

I will review that and submit a PR as soon as possible.
Best regards

**Comment by juan-cb** [comment]
Hi again @kylegao91 

I experienced some issues in the implementation when using `other['topk_sequence']`. In the documentation, it is stated that the `topk_sequence` is a 
> list of beam search sequences, each beam is a list of token IDs


After some tests, I defined the following method:

```
def predict_n(self, src_seq, n=None):
	src_id_seq = Variable(torch.LongTensor([self.src_vocab.stoi[tok] for tok in src_seq]),volatile=True).view(1, -1)
	if torch.cuda.is_available():
		src_id_seq = src_id_seq.cuda()

	softmax_list, _, other = self.model(src_id_seq, [len(src_seq)])
	length = other['length'][0]

	result = []
	for x in range(0, int(n)):
		tgt_id_seq = [other['topk_sequence'][di][0].data[0] for di in range(length)]
		tgt_seq = [self.tgt_vocab.itos[tok] for tok in tgt_id_seq[x]]
		result.append(tgt_seq)
	
	return result
```

Which is used as following:

```
def get_n_responses(question, n):
    get_seq2seq, get_input_vocab, get_output_vocab = load_rnn_from_last_checkpoint()
    beam_search = Seq2seq(get_seq2seq.encoder, TopKDecoder(get_seq2seq.decoder, int(n)))
    predictor = Predictor(beam_search, get_input_vocab, get_output_vocab)
    quest = str(question).strip().split()
    response = predictor.predict_n(quest, n)
```

The problem is that, when I use try to predict the first 6 possible sequences using the example in the readme "1 3 5 7 9"  by this way`get_n_responses("1 3 5 7 9", 6)`, the method only return me an unique response (correct, but using lists within the result list): `[['9'], ['7'], ['5'], ['3'], ['1'], ['<eos>']]`

I think I'm missing something important, and probably I do not use properly the `topk_sequence` list.  Could you shed some light on this?

Thanks

**Comment by kylegao91** [comment]
Hi @juan-cb, sorry for the late reply.  Please find my changes to your `predict_n` below.  It should work, and let's make a pr.

```
    def predict_n(self, src_seq, n=None):
        src_id_seq = Variable(torch.LongTensor([self.src_vocab.stoi[tok] for tok in src_seq]),volatile=True).view(1, -1)
        if torch.cuda.is_available():
            src_id_seq = src_id_seq.cuda()

        softmax_list, _, other = self.model(src_id_seq, [len(src_seq)])

        result = []
        for x in range(0, int(n)):
            length = other['topk_length'][0][x]
            tgt_id_seq = [other['topk_sequence'][di][0,x,0].data[0] for di in range(length)]
            tgt_seq = [self.tgt_vocab.itos[tok] for tok in tgt_id_seq]
            result.append(tgt_seq)

        return result
```

**Comment by juan-cb** [comment]
Hi @kylegao91 

Thank you for reviewing the code and for the fix. I merged your code to mine and submitted the PR #116 

Best regards

**Comment by kylegao91** [comment]
Fixed in #116
